market/listeners.py

```diff
@@ -0,0 +1,34 @@
+__author__ = 'chris'
+import json
+from interfaces import MessageListener
+from zope.interface import implements
+from db.datastore import MessageStore
+from twisted.internet import task, reactor
+
+class MessageListenerImpl(object):
+    implements(MessageListener)
+
+    def __init__(self, web_socket_factory):
+        self.ws = web_socket_factory
+        self.db = MessageStore()
```

Comment:
hmm, so here I suppose. I guess you silently want it to fail on every case after the time it's first created.